src/D2L.Bmx/Aws/AwsRoleState.cs

```diff
@@ -0,0 +1,17 @@
+namespace D2L.Bmx.Aws;
+
+public interface IRoleState {
+	string[] Roles { get; }
+}
+
+public class AwsRoleState : IRoleState {
```

Comment:
I don't see the need for an interface here. the "role state" is just a data object right? so record with primary constructor?